uitest/WidgetTest.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using DevHome.UITest.Common;
+using DevHome.UITest.Dialogs;
+using DevHome.UITest.Pages;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace DevHome.Tests.UITest;
+[TestClass]
+public class WidgetTest : DevHomeTestBase
+{
+    [TestMethod]
+    public void RemoveWidgetTest()
+    {
+        // Arrange
+        var dashboard = Application.NavigateToDashboardPage();
+        dashboard.RemoveAllWidgets();
+
+        // Act
+        var initialWidgetCount = dashboard.DisplayedWidgets.Count;
+        dashboard.ClickAddWidgetButton().AddNetworkUsageWidget();
+        var oneWidgetCount = dashboard.DisplayedWidgets.Count;
+        dashboard.RemoveAllWidgets();
+        var finalWidgetCount = dashboard.DisplayedWidgets.Count;
+
+        // Assert
+        Assert.AreEqual(0, initialWidgetCount);
+        Assert.AreEqual(1, oneWidgetCount);
+        Assert.AreEqual(0, finalWidgetCount);
+    }
```

Comment:
Nit: I don't like creating these variables to evaluate at the end, and I don't see any reason to do it except to stick to the Arrange/Act/Assert pattern.
I think in UI tests it may make more sense to make the assertions at each step instead of all at the end like in the unit tests